sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -711,7 +715,7 @@ nonReserved
     | ADD
     | OVER | PARTITION | RANGE | ROWS | PRECEDING | FOLLOWING | CURRENT | ROW | LAST | FIRST | AFTER
     | MAP | ARRAY | STRUCT
-    | LATERAL | WINDOW | REDUCE | TRANSFORM | USING | SERDE | SERDEPROPERTIES | RECORDREADER
```

Comment:
I found this `USING` entry caused a failure in `PlanParserSuite`;
https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/PlanParserSuite.scala#L336
```
- joins *** FAILED ***
  == FAIL: Plans do not match ===
   'Project [*]                               'Project [*]
  !+- 'Join Inner                             +- 'Join UsingJoin(Inner,List(a, b))
      :- 'UnresolvedRelation `t`                 :- 'UnresolvedRelation `t`
  !   +- 'SubqueryAlias using                    +- 'UnresolvedRelation `u`
  !      +- 'UnresolvedRelation `u`, [a, b] (PlanTest.scala:97)
```